addons/lunch/models/lunch_alert.py

```diff
@@ -84,36 +87,105 @@ def _search_available_today(self, operator, value):
+        for alert in self:
+            alert = alert.with_context(tz=alert.tz)
+
+            cron_required = (
+                alert.active
+                and alert.mode == 'chat'
+                and (not alert.until or fields.Date.context_today(alert) <= alert.until)
+            )
+
+            sendat_tz = pytz.timezone(alert.tz).localize(datetime.combine(
+                fields.Date.context_today(alert, fields.Datetime.now()),
+                float_to_time(alert.notification_time, alert.notification_moment)))
+            lc = alert.cron_id.lastcall
+            if ((lc and sendat_tz.date() <= fields.Datetime.context_timestamp(alert, lc).date())
+                or (not lc and sendat_tz <= fields.Datetime.context_timestamp(alert, fields.Datetime.now()))):
+                sendat_tz += timedelta(days=1)
+            sendat_utc = sendat_tz.astimezone(pytz.UTC).replace(tzinfo=None)
+
+            alert.cron_id.name = f"Lunch: alert chat notification ({alert.name})"
+            alert.cron_id.active = cron_required
+            alert.cron_id.nextcall = sendat_utc
+            alert.cron_id.code = dedent(f"""\
+                # This cron is dynamically controlled by {self._description}.
+                # Do NOT modify this cron, modify the related record instead.
+                env['{self._name}'].browse([{alert.id}])._notify_chat()""")
+
+
+    @api.model
+    def create(self, vals_list):
+        if isinstance(vals_list, dict):
+            vals_list = [vals_list]
+
+        crons = self.env['ir.cron'].sudo().create([
+            {
+                'user_id': self.env.ref('base.user_root').id,
+                'active': False,
+                'interval_type': 'days',
+                'interval_number': 1,
+                'numbercall': -1,
+                'doall': False,
+                'name': "Lunch: alert chat notification",
+                'model_id': self.env['ir.model']._get_id(self._name),
+                'state': 'code',
+                'code': "",
+            }
+            for _ in range(len(vals_list))
+        ])
+        for vals, cron in zip(vals_list, crons):
+            vals['cron_id'] = cron.id
 
-                if alert.location_ids.ids:
-                    order_domain = expression.AND([order_domain, [('user_id.last_lunch_location_id', 'in', alert.location_ids.ids)]])
+        alerts = super().create(vals_list)
+        alerts._sync_cron()
+        return alerts
 
-                if alert.recipients != 'everyone':
-                    weeks = 1
+    def write(self, values):
+        super().write(values)
 
-                    if alert.recipients == 'last_month':
-                        weeks = 4
-                    else:  # last_year
-                        weeks = 52
+        cron_depends = {
+            'name', 'active', 'mode', 'until',
+            'notification_time', 'notification_moment', 'tz',
+        }
+        if set(values) & cron_depends:
```

Comment:
`cron_depends` should be a constant outside this function, and
testing set intersection does not require to compute the intersection:
```suggestion
        if not CRON_DEPENDS.isdisjoint(values):
```

Reply:
You can also use the old-fashioned readable way
```suggestion
        if any(name in values for name in cron_depends):
```